### PR TITLE
extend monitoring

### PR DESCRIPTION
**Pull request by Davka**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,8 +71,12 @@ Also check [Before you install](https://docs.bigbluebutton.org/2.2/install.html#
 | `bbb_monitoring_all_in_one_grafana` | Enable(true)/Disable(false) the Grafana container | `true` | |
 | `bbb_monitoring_all_in_one_prometheus` | Enable(true)/Disable(false) the prometheus container | `true` | |
 | `bbb_monitoring_all_in_one_external` | Enable exposure to nginx | `false` | Can be reached under `/mon/bbb` and `/mon/node` - requires `htpasswd` and `htpasswd_user` |
+| `bbb_monitoring_metrics_endpoint` | Metrics endpoint | `/mon/node/` | |
+| `bbb_monitoring_node_exporter_endpoint` | Node-exporter endpoint
 | `bbb_monitoring_all_in_one_htpasswd_user` | The user for the htpasswd - _(required)_ if external | `Undefined` | |
 | `bbb_monitoring_all_in_one_htpasswd` | The password for the htpasswd - _(required)_ if external | `Undefined` | |
+| `bbb_monitoring_use_nginx_access_module` | Restricting access by IP Address instead using htpasswd | `false` | |
+| `bbb_monitoring_ips` | a list of ip addresses for restricting access | `[]` | in array with key `server` |
```
Comment (n0emis) [inline README.md:79]:
```suggestion
| `bbb_monitoring_ips` | a list of ip addresses for restricting access | `[]` | in array with key `address` |
```

**Review by BrutalBirdie — APPROVED** [review]

File: templates/monitoring/monitoring.nginx.j2 [hunk 2]
```diff
@@ -1,20 +1,36 @@
...
-  auth_basic_user_file /etc/bigbluebutton/nginx/.htpasswd;
+location {{ bbb_monitoring_node_exporter_endpoint }} {
+  proxy_pass http://127.0.0.1:9100/;
+  {% if bbb_monitoring_use_nginx_access_module | bool %}
+{% for item in bbb_monitoring_ips %}
+    allow {{ item.address }};
+{% endfor %}
+    # drop rest of the world
+    deny all;
+  {% else %}
```
Comment (crosscodr) [inline templates/monitoring/monitoring.nginx.j2:30]: I personally wouldn't mind to have both ip restrictions and htpasswd in place.

File: templates/monitoring/monitoring.nginx.j2 [hunk 3]
```diff
@@ -1,20 +1,36 @@
...
+    auth_basic "BigBlueButton Exporter";
+    auth_basic_user_file /etc/bigbluebutton/nginx/.htpasswd;
+  {% endif %}
```
Comment (crosscodr) [inline templates/monitoring/monitoring.nginx.j2:33]:
```suggestion
    {% endif %}
    auth_basic "BigBlueButton Exporter";
    auth_basic_user_file /etc/bigbluebutton/nginx/.htpasswd;
```

File: templates/monitoring/monitoring.nginx.j2 [hunk 4]
```diff
@@ -1,20 +1,36 @@
...
```
Comment (crosscodr) [inline templates/monitoring/monitoring.nginx.j2:18]:
```suggestion
    {% endif %}
    auth_basic "BigBlueButton Exporter";
    auth_basic_user_file /etc/bigbluebutton/nginx/.htpasswd;

```

File: tasks/checkvars.yml [hunk 5]
```diff
@@ -28,11 +28,11 @@
     - name: bbb_monitoring_all_in_one_htpasswd_user
       debug:
         msg: "value: {{ bbb_monitoring_all_in_one_htpasswd_user }}"
-      when: bbb_monitoring_all_in_one_external
+      when: bbb_monitoring_all_in_one_external and not bbb_monitoring_use_nginx_access_module | bool
```
Comment (crosscodr) [inline tasks/checkvars.yml:31]: see my comment to montoring.nginx.j2

File: tasks/main.yml [hunk 6]
```diff
@@ -30,13 +30,13 @@
...
 - import_tasks: docker.yml
-  when: bbb_greenlight_enable | bool or bbb_monitoring_all_in_one_enable | bool
+  when: bbb_greenlight_enable | bool or bbb_monitoring_all_in_one_enable | bool or (bbb_monitoring_all_in_one_external | bool)
 
 - import_tasks: greenlight.yml
   when: bbb_greenlight_enable | bool
 
 - import_tasks: monitoring/all_in_one.yml
-  when: bbb_monitoring_all_in_one_enable | bool
+  when: (bbb_monitoring_all_in_one_enable | bool) or (bbb_monitoring_all_in_one_external | bool)
```
Comment (crosscodr) [inline tasks/main.yml:39]: I don't understand why do you need this here?

**Review by crosscodr — CHANGES_REQUESTED** [review]

File: tasks/checkvars.yml (hunk 5 above)
Comment (Davka) [reply tasks/checkvars.yml:31]: I changed it

**Comment by crosscodr** [comment]
Sorry @Davka, this PR has to be reworked again.
There are a few simplifications made in #134, e.g. unified variable names.
Please merge and update this PR.

**Comment by Davka** [comment]
Will submit a new PR

**Pull request closed without merging**